standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -1740,4039 +649,413 @@ public long getLatestTxnIdInConflict(long txnid) throws MetaException {
-              JavaUtils.lockIdToString(extLockId), String.join(", ", notFoundIds), JavaUtils.txnIdToString(txnId));
-      throw new NoSuchLockException(errorMsg);
-    }
+  public MetricsInfo getMetricsInfo() throws MetaException {
+    int threshold = MetastoreConf.getIntVar(conf, MetastoreConf.ConfVars.METASTORE_ACIDMETRICS_TABLES_WITH_ABORTED_TXNS_THRESHOLD);
+    MetricsInfo metrics = jdbcResource.execute(MetricsInfoHandler.INSTANCE);
+    Set<String> resourceNames = jdbcResource.execute(new TablesWithAbortedTxnsHandler(threshold));
+    metrics.setTablesWithXAbortedTxnsCount(resourceNames.size());
+    metrics.setTablesWithXAbortedTxns(resourceNames);    
+    return metrics;
   }
 
-  /**
-   * Heartbeats on the lock table.  This commits, so do not enter it with any state.
-   * Should not be called on a lock that belongs to transaction.
-   */
-  private void heartbeatLock(Connection dbConn, long extLockId)
-    throws NoSuchLockException, SQLException, MetaException {
-    // If the lock id is 0, then there are no locks in this heartbeat
-    if (extLockId == 0) {
-      return;
-    }
-    try (Statement stmt = dbConn.createStatement()) {
-      String updateHeartbeatQuery = "UPDATE \"HIVE_LOCKS\" SET \"HL_LAST_HEARTBEAT\" = " +
-          getEpochFn(dbProduct) + " WHERE \"HL_LOCK_EXT_ID\" = " + extLockId;
-      LOG.debug("Going to execute update <{}>", updateHeartbeatQuery);
-      int rc = stmt.executeUpdate(updateHeartbeatQuery);
-      if (rc < 1) {
-        LOG.error("Failure to update last heartbeat for extLockId={}.", extLockId);
-        dbConn.rollback();
-        throw new NoSuchLockException("No such lock: " + JavaUtils.lockIdToString(extLockId));
-      }
-      LOG.debug("Successfully heartbeated for extLockId={}", extLockId);
-      dbConn.commit();
-    }
-  }
 
-  // Heartbeats on the txn table.  This commits, so do not enter it with any state
-  private void heartbeatTxn(Connection dbConn, long txnid)
-    throws NoSuchTxnException, TxnAbortedException, SQLException, MetaException {
-    // If the txnid is 0, then there are no transactions in this heartbeat
-    if (txnid == 0) {
-      return;
-    }
-    try (Statement stmt = dbConn.createStatement()) {
-      String s = "UPDATE \"TXNS\" SET \"TXN_LAST_HEARTBEAT\" = " + getEpochFn(dbProduct) +
-          " WHERE \"TXN_ID\" = " + txnid + " AND \"TXN_STATE\" = " + TxnStatus.OPEN;
-      LOG.debug("Going to execute update <{}>", s);
-      int rc = stmt.executeUpdate(s);
-      if (rc < 1) {
-        ensureValidTxn(dbConn, txnid, stmt); // This should now throw some useful exception.
-        LOG.error("Can neither heartbeat txn (txnId={}) nor confirm it as invalid.", txnid);
-        dbConn.rollback();
-        throw new NoSuchTxnException("No such txn: " + txnid);
-      }
-      LOG.debug("Successfully heartbeated for txnId={}", txnid);
-      dbConn.commit();
-    }
+  private static void shouldNeverHappen(long txnid) {
```

Comment:
remove extra new line above and maybe move all private methods to the end of the class

Reply:
done